### PR TITLE
Remove etcd snapshot migration code

### PR DESCRIPTION
**Pull request by calvix**
towards https://github.com/giantswarm/giantswarm/issues/9979 

it should also speed up the cluster creation by a lot

unfortunately, the `etcdnsnapshot`  code needs to stay for cluster created by it, as if we would remove the snapshot ID for the already created etcdVolume, the `tccpn` stack would fail

### CODE REVIEW

File: service/controller/resource/tccp/create.go [hunk 1]
```diff
@@ -117,44 +114,13 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
-			err = r.snapshotEtcdVolume(ctx, cr)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
-			err = r.terminateMasterInstance(ctx, cr)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
```
Comment (calvix) [inline service/controller/resource/tccp/create.go:139]: master instance  in tccp is gone no need for this code anymore as well

File: service/controller/resource/tccp/create.go [hunk 2]
```diff
@@ -117,44 +114,13 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
-	// https://github.com/giantswarm/giantswarm/issues/9979
-	if key.IsNewCluster(cr) && cc.Status.TenantCluster.MasterInstance.EtcdVolumeSnapshotID == "" {
-		err = r.snapshotEtcdVolume(ctx, cr)
-		if err != nil {
-			return microerror.Mask(err)
-		}
-
-		r.logger.LogCtx(ctx, "level", "debug", "message", "created etcd volume snapshot for newly created cluster")
-	}
-
```
Comment (calvix) [inline service/controller/resource/tccp/create.go:157]: removing the TODO code

File: service/controller/resource/tccp/create.go [hunk 3]
```diff
@@ -319,38 +233,6 @@ func (r *Resource) newParamsMain(ctx context.Context, cr infrastructurev1alpha2.
...
-			Master: template.ParamsMainInstanceMaster{
-				/*
-					This is part of migration code we introduced with the new tccpn stack.
-					We need this to create empty ETCD volume in the first AZ  to do a snapshot
-					and then create volume from that snapshot in the tccpn stack.
-
-					We can take a look at how we could remove it before creating stable release for HA masters.
-				*/
-				AZ: haMapping[0].AZ,
-				EtcdVolume: template.ParamsMainInstanceMasterEtcdVolume{
```
Comment (calvix) [inline service/controller/resource/tccp/create.go:345]: we don't need the etcd volume in tccp anymore, so removing the whole template that was left there

File: service/controller/resource/tccpn/create.go [hunk 4]
```diff
@@ -47,12 +47,6 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 			return nil
 		}
 
-		if cc.Status.TenantCluster.MasterInstance.EtcdVolumeSnapshotID == "" {
-			r.logger.LogCtx(ctx, "level", "debug", "message", "etcd volume snapshot id not available yet")
-			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-			return nil
-		}
-
```
Comment (calvix) [inline service/controller/resource/tccpn/create.go:55]: tccpn does not need to wait for that value anymore as the new cluster will be without `etcdsnapshot`

and for older cluster this  will be properly set as the snapshot is created and the resource that is fetching the id will be executed before this code

File: service/controller/cluster.go [hunk 5]
```diff
@@ -48,7 +48,6 @@ import (
 	"github.com/giantswarm/aws-operator/service/controller/resource/s3bucket"
 	"github.com/giantswarm/aws-operator/service/controller/resource/secretfinalizer"
 	"github.com/giantswarm/aws-operator/service/controller/resource/service"
-	"github.com/giantswarm/aws-operator/service/controller/resource/snapshotid"
```
Comment (njuettner) [inline service/controller/cluster.go:51]: The snapshotid handler is this still necessary for the control plane? or is also droppable?

Comment (calvix) [reply service/controller/cluster.go:51]: as written here https://github.com/giantswarm/aws-operator/pull/2596#issue-449306245

due to the old cluster having it in CF stack we cannot simply drop it

Comment (njuettner) [reply service/controller/cluster.go:51]: I was not reading it carefully enough

**Comment by calvix** [comment]
list of nodepool clusters with release < 11.2.0
```
iris eg4uz,
icarus 04wre
viking  yz2cx, ktc6d, 7fkvu
```

aparently iris and icarus will be updated this weekend and the viking clusters will be updated next week SO I will park it here and  check abck end of next week.

**Comment by xh3b4sd** [comment]
With the latest changes around the k8s deps to 1.18 we would release a new major version `v9.0.0`. Can this change here be merged now?

**Comment by calvix** [comment]
> With the latest changes around the k8s deps to 1.18 we would release a new major version v9.0.0. Can this change here be merged now?

I was looking at the versions last week and there was still one cluster with the old version, but maybe this changed. I will recheck today and see if I can merge.

**Comment by calvix** [comment]
there is still
```
yz2cx   digital-pt            digital-pt-prd         11.1.4    2020 Jan 09, 07:10 UTC
```

that cluster needs to go over some version   11.2.0 >= x <= 12.1.1 otherwise it could break :( I will ask in vodafone channel

**Comment by xh3b4sd** [comment]
My thinking was we can merge this into master since the latest version is a new major and we have an active > `v11.5.x`.

**Comment by calvix** [comment]
tested few creation/updates and  i could not see any issue,  s same result as i was testing before.

Merging

**Pull request merged**